org/mozilla/jss/ssl/javax/JSSParameters.java

```diff
@@ -0,0 +1,174 @@
+        // everywhere to rely on the default NSS policies.
+        super(null, null);
+
+        suites = null;
+        range = null;
+    }
+
+    public JSSParameters(SSLParameters downcast) {
+        this();
+        if (downcast == null) {
+            return;
+        }
+
+        // Copy data from downcast
+        setCipherSuites(downcast.getCipherSuites());
+        setProtocols(downcast.getProtocols());
+        setAlgorithmConstraints(downcast.getAlgorithmConstraints());
+        setEndpointIdentificationAlgorithm(downcast.getEndpointIdentificationAlgorithm());
+        setNeedClientAuth(downcast.getNeedClientAuth());
+        setServerNames(downcast.getServerNames());
+        setSNIMatchers(downcast.getSNIMatchers());
+        setUseCipherSuitesOrder(downcast.getUseCipherSuitesOrder());
+        setWantClientAuth(downcast.getWantClientAuth());
+    }
+
+    public JSSParameters(String[] cipherSuites) {
+        super(cipherSuites);
+        setCipherSuites(cipherSuites);
+    }
+
+    public JSSParameters(String[] cipherSuites, String[] protocols) {
+        super(cipherSuites, protocols);
+        setCipherSuites(cipherSuites);
+        setProtocols(protocols);
+    }
+
+    public void setCipherSuites(String[] cipherSuites) throws IllegalArgumentException {
+        if (cipherSuites == null || cipherSuites.length == 0) {
+            suites = null;
+            return;
+        }
+
+        ArrayList<SSLCipher> converted = new ArrayList<SSLCipher>();
+        for (String cipherSuite : cipherSuites) {
+            try {
+                converted.add(SSLCipher.valueOf(cipherSuite));
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Unknown cipher suite (" + cipherSuite + "): " + e.getMessage(), e);
+            }
+        }
+
+        suites = converted.toArray(new SSLCipher[0]);
+    }
+
+    public void setCipherSuites(AbstractList<SSLCipher> cipherSuites) {
+        setCipherSuites(cipherSuites.toArray(new SSLCipher[0]));
+    }
+
+    public void setCipherSuites(SSLCipher[] cipherSuites) {
+        if (cipherSuites == null || cipherSuites.length == 0) {
+            suites = null;
+            return;
+        }
+
+        // Perform a copy of cipherSuites in case it is modified later.
+        suites = new SSLCipher[cipherSuites.length];
+        for (int index = 0; index < cipherSuites.length; index++) {
+            suites[index] = cipherSuites[index];
+        }
+    }
+
+    public String[] getCipherSuites() {
+        if (suites == null) {
+            return null;
+        }
+
+        ArrayList<String> ciphers = new ArrayList<String>();
+        for (SSLCipher suite : suites) {
+            ciphers.add(suite.name());
+        }
+
+        return ciphers.toArray(new String[0]);
+    }
+
+    public SSLCipher[] getSSLCiphers() {
+        return suites;
+    }
+
+    public void setProtocols(String[] protocols) {
```

Comment:
Should probably add a `throws IllegalArgumentException` here.

Reply:
OK